xWebAppPool updates

SNikalaichyk opened the pull request:
* Bug fixes, error handling and input validation improvements.
* The resource was updated to ensure a specific state only for the explicitly specified properties.
* The following properties were added: **idleTimeoutAction**, **logEventOnProcessModel**, **setProfileEnvironment**.
* The type of the following properties was changed to **Boolean**: **autoStart**, **enable32BitAppOnWin64**, **enableConfigurationOverride**,
    **passAnonymousToken**, **cpuSmpAffinitized**, **loadUserProfile**, **manualGroupMembership**, **pingingEnabled**, **setProfileEnvironment**,
    **orphanWorkerProcess**, **rapidFailProtection**, **disallowOverlappingRotation**, **disallowRotationOnConfigChange**.
* Unit and integration tests updated.
* Added sample configuration.
* Detailed property descriptions in **README.md**.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xwebadministration/98)
<!-- Reviewable:end -->

pagnmickie commented:
When will this be merged? I'm seeing this same issue.

andrewwipf commented:
Are there any workarounds to #96 until this gets merged?

tysonjhayes commented:
I'm just having a chance at looking at this. Your compounding a bunch of changes together that I think we may need to revert and submit as separate PRs.  This is going to take me a while to unravel and get merged.

The first thing I'm going to ask broadly is why the change to the boolean for most of the properties? I believe that is going to be a schema breaking change which I don't want to do lightly, I also feel that most of that type of logic change should be done on it's own PR if possible.

Second point is you are doing a number of style changes that go against our current style. For example you add a new line in all of our parameters next to their type. Please revert these as it should be ````[string] $Name```` not
````
[string]
$Name
````

Third you do a lot of validation to make sure that we are between a set of numbers ie:

````powershell
[ValidateScript({([ValidateRange(0, 432000)]$ValueInMinutes = [TimeSpan]::Parse($_).TotalMinutes); $?})] # The valid range (in minutes) is 0 to 432000.
````

I'm not following why we are are validating the range and then running the parse on the value. Why isn't ValidateRange here sufficient?

I'll need to spend some more time on this but if we could remove most of the stylistic changes and just get this down to some of the logic changes it would go a long way to speeding up my ability to review this.

tysonjhayes commented:
Review status: 0 of 7 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 109 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3W5GS8aq6z36bSsoO:-KE3W5GS8aq6z36bSsoP:-1984361799) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L109)):*
Not really sure if this comment is adding any real value. If you want to keep it could you move it above the parameter please?

---

*[Tests/Integration/MSFT_xWebAppPool.config.ps1, line 18 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3hQtPJB5h3xKY2FKB:-KE3hQtQ5hCQn4hukhoY:-405737602) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/Tests/Integration/MSFT_xWebAppPool.config.ps1#L18)):*
I'm not seeing the value in renaming this, this seems like an unnecessary change to me for this PR. If you want to change the readability of this please submit as a separate PR.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/powershell/xwebadministration/98)*
<!-- Sent from Reviewable.io -->

tysonjhayes commented:
Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 106 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3pBc-ErRrA3WNonmG:-KE3pBc-ErRrA3WNonmH:359768148) ([raw file](https://github.com/powershell/xwebadministration/blob/ec053506ebcadc6f64801e6dc951f24fcd060c03/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L106)):*
Could you help me understand why you removed this?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 117 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3oVjxWzvFlGC4tm6k:-KE3oVjxWzvFlGC4tm6l:-1638804839) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L117)):*
Can you help me understand the value in this change? 

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 119 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3okdhKS5DVOvN49zJ:-KE3okdhKS5DVOvN49zK:871523839) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L119)):*
I'm thinking if we move the $cimCredential = $null and $ensureResult = 'Absent' here we could remove a few if/else blocks from this and make the logic a bit easier to follow.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 120 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p1t62i18XRAsqb5F:-KE3p1t62i18XRAsqb5G:1809476702) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L120)):*
Effectively we could remove this if block.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 143 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p6A3tcdJkcxAuI-a:-KE3p6A3tcdJkcxAuI-b:-343731888) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L143)):*
Also means this else block is unnecessary.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 325 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3pijYm9s9rj6dW6SY:-KE3pijYm9s9rj6dW6SZ:-477521881) ([raw file](https://github.com/powershell/xwebadministration/blob/ec053506ebcadc6f64801e6dc951f24fcd060c03/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L325)):*
Could you help me understand why you want to remove this?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 469 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3qQXxxLa0miFUYMSw:-KE3qQXxxLa0miFUYMSx:-1064715178) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L469)):*
I have a personal pet peeve of elseifs without an else. I'm assuming the else case would be '<=>' or something to that effect? Could we add an else that basically says no change is needed or something?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/powershell/xwebadministration/98)*
<!-- Sent from Reviewable.io -->

SNikalaichyk commented:
Hello Tyson!
Could you please just treat all the changes in this PR as a major release? I simply don't have time to split this PR into a number of smaller ones. The current version of **xWebAppPool** is botched and needs to be overhauled. I've done it and this is the result.

*Booleans*:
Property types were changed to match types returned by the `Get-WebConfiguration` cmdlet and what is defined in the [configuration reference](https://www.iis.net/configreference/system.applicationhost/applicationpools/add). I also believe that `IsPropertyEnabled = $true` syntax is more natural to PowerShell than `IsPropertyEnabled = 'true'`.

*Parameter declaration style*: 
The following approach is a lot more readable in case there is a whole bunch of parameters. I would recommend to update the styling guide to make this a standard.
```powershell
[String]
$ParameterName
```

*Timespan validation*:

I accomplish two tasks here:
    1. Validate the string is a valid TimeSpan string (`[TimeSpan]::Parse`).
    2. Validate TimeSpan value range (`[ValidateRange()]`).

`[ValidateRange()]` is used inside the `[ValidateScript()]` because it is super fast:
```powershell
# Case 1 (1500 times faster than Case 2)
1..100 | ForEach-Object {

    Measure-Command -Expression {

        [ValidateScript({
            ([ValidateRange(1, 4294967)]$ValueInSeconds = [TimeSpan]::Parse($_).TotalSeconds); $?
        })][String]$shutdownTimeLimit = '14.00:00:00'

    }

} | Measure-Object -Property TotalMilliseconds -Average

# Case 2
1..100 | ForEach-Object {

    Measure-Command -Expression {

        [ValidateScript({
            [TimeSpan]::Parse($_).TotalSeconds -in @(1..4294967)
        })][String]$shutdownTimeLimit = '14.00:00:00'

    }

} | Measure-Object -Property TotalMilliseconds -Average

```
It also produces a nice error message in case the specified value falls out of the allowed range of values. Please find the example below:
```
PowerShell DSC resource MSFT_xWebAppPool  failed to execute Test-TargetResource functionality with error message: Cannot validate argument on parameter 'shutdownTimeLimit'. The attribute cannot be added because variable ValueInSeconds with value 4294968 would no longer be valid.
```

Please feel free to reach me by Skype. I will gladly answer all of your questions.

---

Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 106 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3pBc-ErRrA3WNonmG:-KEB37BYLgxSRzlJWH-o:215832143) ([raw file](https://github.com/powershell/xwebadministration/blob/ec053506ebcadc6f64801e6dc951f24fcd060c03/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L106)):*
It is redundant because it is not possible to have multiple identically named application pools.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 109 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3W5GS8aq6z36bSsoO:-KEB8x3dOd_HOgkHN9yt:-663491029) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L109)):*
I agree that it's self-explanatory. Will remove this comment.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 117 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3oVjxWzvFlGC4tm6k:-KEB9TPkxkRY1-FjeXwE:-1157122500) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L117)):*
This specific change is the most important one.

`AppCmd.exe` in the following line returns raw text:
```powershell
[xml] $PoolConfig = Invoke-AppCmd -Arguments list,apppool,$Name,/config:*
```
This approach is prone to errors.

`Get-WebConfiguration` returns proper PowerShell objects that do not require casting/parsing. Moreover this is more natural to PowerShell.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 119 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3okdhKS5DVOvN49zJ:-KEBD9XZJmQkpB_NYYz8:-1716102468) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L119)):*
Not sure I understand exactly what you mean here. Please clarify.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 143 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p6A3tcdJkcxAuI-a:-KEBDosxvrOGDvFDiolR:-1726853205) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L143)):*
Initially I wanted to implement support for Strict Mode. That was the reason to have this `else` block. Later I abandoned this idea because it required a lot more effort than I had thought before. I can remove it, though it may be useful for enabling strict mode in future.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 325 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3pijYm9s9rj6dW6SY:-KEBMzts129XNaHyRw_y:-2057573096) ([raw file](https://github.com/powershell/xwebadministration/blob/ec053506ebcadc6f64801e6dc951f24fcd060c03/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L325)):*
Diff is not displayed properly. Could you please clarify?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 469 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3qQXxxLa0miFUYMSw:-KEB6jgso-76E7COiIV2:-340337985) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L469)):*
The value of `==` is technically possible if someone adds the `-IncludeEqual` parameter. Having `elseif` will always ensure the code works as expected.

---

*[Tests/Integration/MSFT_xWebAppPool.config.ps1, line 18 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3hQtPJB5h3xKY2FKB:-KEB4-VSG-BXKNPPGvg2:1486917518) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/Tests/Integration/MSFT_xWebAppPool.config.ps1#L18)):*
I personally think that `$TestParameters` is more self-descriptive than `$rules`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KEBPAjdOV27Xi7sQiDm:-484813049)*
<!-- Sent from Reviewable.io -->

SNikalaichyk commented:
Hi, any updates on this?

PlagueHO commented:
Hi @SNikalaichyk, I'm just taking a look through this to see what needs to be done to get this merged for you.

It looks like there are a few changes in here pushed into a single PR which makes review a bit tricker (my poor brain can't keep up :wink: ). It would definitely help to separate large changes like this into smaller PR's - only to make it a bit simpler to review.

Because some of the StyleGuidelines are in the process of being implemented as part of the standard tests it is important for every PR to meet them so that when the tests are in place each Module doesn't require lots of changes to allow the tests to begin passing again.

Also, I've raised an issue in the DSCResources repo to see if we can get a final resolution on the parameter format. Feel free to head over to the issue and weigh in (https://github.com/PowerShell/DscResources/issues/121)

I myself prefer the ```[string] $name``` format because that's the way I declare variables in the code block and it does allow the param block to be shrunk to some degree.

Anyway, hopefully my review can help get things moving. If you can get the other stuff done while we hear about the StyleGuidelines then that will get this moving forward. I'll review the changes again once you've had a change to look at my comments.

Thanks for your submission though - it looks like it contains a lot of valuable functionality (just makes me go cross-eyed trying to review it) :smile:

---

Review status: 0 of 7 files reviewed at latest revision, 33 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 109 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3W5GS8aq6z36bSsoO:-KFH4mQazPP8_jq-tD1k:1596326367) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L109)):*
This comment hasn't been removed yet :smile: 

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 119 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3okdhKS5DVOvN49zJ:-KFH5-ZD8Ktyys_EfNgg:-629543934) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L119)):*
I think I understand:
Could you move the
```powershell
$cimCredential = $null
```
To before the
```powershell
 if ($appPool -eq $null)
```

block.
This will enable ```$cimCredential = $null``` to be removed from the two places further down.

I think the ```$ensureResult = 'Absent' ``` could stay though.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 120 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p1t62i18XRAsqb5F:-KFH6xL5-bAGFC2BXA2o:1545109129) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L120)):*
Although I guess if this block is kept then the Verbose logs can report the "VerboseAppPoolNotFOund" vs "VerboseAppPoolFound" - might be helpful for debugging.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 143 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p6A3tcdJkcxAuI-a:-KFH6OAvGYCD4Lzf5hWo:1145691623) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L143)):*
If you set ```$cimCredential = $null``` at the beginning of the function then this becomes redundant anyway (strict mode or not) :smile: The whole else block could go.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 180 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7VOocxxluCTP98Qg:-KFH7VOocxxluCTP98Qh:2011399165) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L180)):*
Could we remove this comment please?

I think with a resource of this size we should try and do everything possible to keep it as slim as possible.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 222 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7fcp-9AijwUeMI-j:-KFH7fcp-9AijwUeMI-k:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L222)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 230 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7lwLmLBYZFjmhw5i:-KFH7lwLmLBYZFjmhw5j:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L230)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 275 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7ybGuyfJnMvPn1Lf:-KFH7ybHY9igrhFMiRHQ:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L275)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 297 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH9Lk4Yv8zj7v-65FE:-KFH9Lk4Yv8zj7v-65FF:-131894578) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L297)):*
With these [ValidateScript] decorations is there anyway to get them down below 100 chars?

E.g.
```powershell
[ValidateScript({
  ([ValidateRange(1, 4294967)]$ValueInSeconds = [TimeSpan]::Parse($_).TotalSeconds); $?
})] # The valid range (in seconds) is 1 to 4294967.
```

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 321 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7t6Ju9hKBhM_IwVx:-KFH7t6Ju9hKBhM_IwVy:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L321)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 340 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH80UcRw3eQsOxMFfT:-KFH80UcRw3eQsOxMFfU:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L340)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 343 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH81SEkjgcFWtIY2Ia:-KFH81SEkjgcFWtIY2Ib:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L343)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 346 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH82pjBeW50bEilOf3:-KFH82pjBeW50bEilOf4:-1804098380) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L346)):*
Could we remove this comment too please?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 376 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHAKu8E9JpI-rqNZpM:-KFHAKu8E9JpI-rqNZpN:517747747) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L376)):*
What is the purpose of this 5 second pause? If it is to ensure that the Pool has had time to be created before moving on, is it possible to detect if it has occurred before moving on? Perhaps implement a loop detecting if the pool has actually been created with a max timeout? Might be a bit safer and also faster if the pool is available in less than 5 seconds.

E.g.
```powershell
    [DateTime] $StartTime = Get-Date
    while ((Get-Date) - $StartTime).TotalSeconds) -lt 15)
    {
            if ( The pool is available )
            {
               break
            }
            Start-Sleep -Seconds 1
    } # while
```

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 381 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHA3Yd8Q0jU_igDlUf:-KFHA3Yeep0aSHlmciUq:-989522630) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L381)):*
Can we get this line down to under 100 chars?
e.g.
```powershell
New-TerminatingError `
  -ErrorId 'ErrorNewAppPool' `
  -ErrorMessage $errorMessage `
  -ErrorCategory 'InvalidOperation'
```

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 386 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHCqMSLdw2Ow3Ks7vX:-KFHCqMSLdw2Ow3Ks7vY:1062713654) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L386)):*
Could this be just replaced with an ```else``` instead? No need for ```if ($appPool -ne $null)``` because the previous block is ```if ($appPool -eq $null)```.

e.g.
```powershell
}
else
{
# Set Application Pool Properties
```

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 414 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHEdvYWGVZu1T-Wu5J:-KFHEdvYWGVZu1T-Wu5K:832986977) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L414)):*
Can we get this line down to under 100 chars - e.g.:
```powershell
Write-Verbose -Message ($LocalizedData['VerboseSetProperty'] ` -f
  'Credential (userName)', $Name)
```

In case it helps, I use VSCode to do dev with and I set line wrap to 100 chars so I can easily see when my lines violate the 100 char rule. I think there is a PSSA warning that will also check this as well. So definitely worth keeping in mind.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 415 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFFiJ8_c59IWKl14E:-KFHFFiJ8_c59IWKl14F:647616465) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L415)):*
Can we get this line down to under 100 chars?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 422 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFJadI499oPIC4QdO:-KFHFJaeqe76Cku9Ikxm:647616465) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L422)):*
Can we get this line down to under 100 chars?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 423 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFKmeEPPBWIt0bOCO:-KFHFKmeEPPBWIt0bOCP:647616465) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L423)):*
Can we get this line down to under 100 chars?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 435 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFNVjp8OXrm316RxQ:-KFHFNVjp8OXrm316RxR:-1373969577) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L435)):*
Can we get this line down to under 100 chars?

Suggest something like this:
```powershell
if (
  (
     (($PSBoundParameters.ContainsKey('identityType') -eq $true) `
      -and ($PSBoundParameters['identityType'] -ne 'SpecificUser')) -or `
     (($PSBoundParameters.ContainsKey('identityType') -eq $false) `
      -and ($appPool.processModel.identityType -ne 'SpecificUser'))
   ) -and
   (
     ([String]::IsNullOrEmpty($appPool.processModel.userName) -eq $false) -or
     ([String]::IsNullOrEmpty($appPool.processModel.password) -eq $false)
   )
)
```

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 453 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHI6BPkh2X80qowzWm:-KFHI6BQZmVGni4mvERd:549607291) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L453)):*
Can we get this down to under 100 chars?

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 461 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHIDqP6587xmZTTA2R:-KFHIDqQ2HizR36MECdb:-1915745626) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L461)):*
Can we get this down to under 100 chars?

I won't report any more of these :) But there are a few more.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 469 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3qQXxxLa0miFUYMSw:-KFHIbR-GUeYxc2XaKT7:-181748352) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L469)):*
I agree here, the ```elseif``` actually adds an additional comparison that isn't required - and as we're doing lots of them in a loop if we can trim it down then it will optimize it. Should there be a reason to add the ```-IncludeEqual``` then it can be readdressed I think- Agile dev practices would indicate that if it isn't needed now, don't include it :)

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 482 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHJe9vrrS92-HdzVL1:-KFHJe9zKh9QMM3y6V6F:-401738693) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L482)):*
Is the Try...Catch block actually necessary? Are we not just obfuscating the error? I'm actually guilty of doing this exact thing myself before.

E.g. the whole try...catch can be removed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 495 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHK7dJz9ow8aXOe4cv:-KFHK7dKnkbgoaArCcNa:-401738693) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L495)):*
Is the Try...Catch block actually necessary? Are we not just obfuscating the error? I'm actually guilty of doing this exact thing myself before.

E.g. the whole try...catch can be removed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 518 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHLIbc7GRICw-nmhdG:-KFHLIbdShUWFQ_CBCox:-401738693) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L518)):*
Is the Try...Catch block actually necessary? Are we not just obfuscating the error? I'm actually guilty of doing this exact thing myself before.

E.g. the whole try...catch can be removed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 530 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHLKoc2vntAIAInJnH:-KFHLKoc2vntAIAInJnI:-401738693) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L530)):*
Is the Try...Catch block actually necessary? Are we not just obfuscating the error? I'm actually guilty of doing this exact thing myself before.

E.g. the whole try...catch can be removed.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 554 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHLPW6iWsJ6M6Fk27l:-KFHLPW6iWsJ6M6Fk27m:-1413628094) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L554)):*
See all my comments from the params block of Set-TargetResource

---

*[Tests/Integration/MSFT_xWebAppPool.config.ps1, line 18 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3hQtPJB5h3xKY2FKB:-KFHMbPEOuSzUzzv5qPT:755368280) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/Tests/Integration/MSFT_xWebAppPool.config.ps1#L18)):*
I'm OK with the purpose behind the change. I agree it is more self-descriptive, but keeping the changes smaller and breaking them into separate PRs makes it soooo much easier to review. :) Besides if one change isn't accepted it doesn't hold up the whole PR. Maybe we could accept this one as is, but in future all changes should be broken down into smaller PRs. Personally I'm guilty of a couple of these large changes myself (sorry @tysonjhayes - I know you've had to review some them - my bad).

Plus, this massive change completely cripples Reviewable in my browser (the page is sooo slow).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KFGtQv5bHc5QE2nXxBe:-940467246)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 33 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98)*
<!-- Sent from Reviewable.io -->

SNikalaichyk commented:
*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 386 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHCqMSLdw2Ow3Ks7vX:-KFKMpuoK7CU-LhoXNL1:-1434360896) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L386)):*
This one is by design actually. In case `$appPool` doesn't exist, a new one is created in the `if ($appPool -eq $null)` block, then its properties are set. `$appPool`may or may not exist before the `Set-TargetResource` function execution. The `if ($appPool -ne $null)` block applies to both cases.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98)*
<!-- Sent from Reviewable.io -->

SNikalaichyk commented:
Thanks for the feedback! Please review the changes I've made.

---

Review status: 2 of 7 files reviewed at latest revision, 33 unresolved discussions.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 109 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3W5GS8aq6z36bSsoO:-KFL7rWg9IJOyepIyUbd:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L109)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 119 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3okdhKS5DVOvN49zJ:-KFL7v5MMIrnWJIwGfuS:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L119)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 120 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p1t62i18XRAsqb5F:-KFL7x_Bh5D032pXHJti:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L120)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 143 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3p6A3tcdJkcxAuI-a:-KFL7zdf_zl7i4bbcCvH:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L143)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 180 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7VOocxxluCTP98Qg:-KFL80mMf2T5EbsKOkcv:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L180)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 222 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7fcp-9AijwUeMI-j:-KFL83_qjZRGefEYxMdE:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L222)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 230 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7lwLmLBYZFjmhw5i:-KFL84ZztvkBCpUIUzbZ:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L230)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 275 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7ybGuyfJnMvPn1Lf:-KFL86-TTyJsKoiBDjpv:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L275)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 297 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH9Lk4Yv8zj7v-65FE:-KFL87N7SsfHqm2AQ6zJ:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L297)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 321 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH7t6Ju9hKBhM_IwVx:-KFL88LlaDgLjtSjSVdK:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L321)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 340 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH80UcRw3eQsOxMFfT:-KFL89IbAyrqQKuNPmwb:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L340)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 343 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH81SEkjgcFWtIY2Ia:-KFLAOYYMbTOltCFzB3v:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L343)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 346 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFH82pjBeW50bEilOf3:-KFL8A3r_AdTPtAoNY3y:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L346)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 376 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHAKu8E9JpI-rqNZpM:-KFL8EC_MbxyFuNKaawb:1580327578) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L376)):*
Removed the timeout because the `New-WebAppPool` output is stored in the `$appPool` variable. There's no need to search for the newly created application pool.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 381 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHA3Yd8Q0jU_igDlUf:-KFL91CcUqLYLWjEsb9_:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L381)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 414 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHEdvYWGVZu1T-Wu5J:-KFL92nj3ZieYZZ2sE0k:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L414)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 415 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFFiJ8_c59IWKl14E:-KFL93TRbQL4sAZVfpPK:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L415)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 422 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFJadI499oPIC4QdO:-KFL94sOm2bExGpWwo50:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L422)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 423 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFKmeEPPBWIt0bOCO:-KFL95GJur9c1q9MztWI:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L423)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 435 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHFNVjp8OXrm316RxQ:-KFL96nlHDU12xvl1h2p:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L435)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 453 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHI6BPkh2X80qowzWm:-KFL9BfEFJLk4bmIBCfF:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L453)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 461 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHIDqP6587xmZTTA2R:-KFL9DDLj90wUV2q5zza:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L461)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 469 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3qQXxxLa0miFUYMSw:-KFL9GYT5qWZn3TFegbW:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L469)):*
Done.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 482 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHJe9vrrS92-HdzVL1:-KFL9I4-dKZds4yvQG6J:-462506902) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L482)):*
Removed try/catch blocks.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 495 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHK7dJz9ow8aXOe4cv:-KFL9Tl35NPe_Wcl8PV8:-462506902) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L495)):*
Removed try/catch blocks.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 518 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHLIbc7GRICw-nmhdG:-KFL9Vf3daCfopFt_S_l:-462506902) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L518)):*
Removed try/catch blocks.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 530 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHLKoc2vntAIAInJnH:-KFL9X_ZteCP-KqleE9o:-462506902) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L530)):*
Removed try/catch blocks.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 554 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHLPW6iWsJ6M6Fk27l:-KFL9ZjSHm5S5j32jITw:-499146009) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L554)):*
Done.

---

*[Tests/Integration/MSFT_xWebAppPool.config.ps1, line 18 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3hQtPJB5h3xKY2FKB:-KFL9lefCD1vF66O8E8m:-1904482944) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/Tests/Integration/MSFT_xWebAppPool.config.ps1#L18)):*
Sorry guys. I know it's a tedious task. Next time.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KFLAQVaeu_fKdFHfStO:-347066720)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Awesome! :) Thank you - Really appreciate all your work on this - you did well getting through all my comments. Glad we have Reviewable now to help track this stuff.

Everything is looking pretty good to me- but we'll wait to get a decision on the Style Guidelines issues I put through - hopefully we can sort all that out once and for all. :)

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1, line 386 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KFHCqMSLdw2Ow3Ks7vX:-KFM8k2NbzfF0ISWhXUk:1681403243) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1#L386)):*
Ah I see! You're quite right. Sorry about that.

---

*[Tests/Integration/MSFT_xWebAppPool.config.ps1, line 18 \[r1\]](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-KE3hQtPJB5h3xKY2FKB:-KFM9P8cIvC5Cyg1GNzL:-329790744) ([raw file](https://github.com/powershell/xwebadministration/blob/a8f588a71e2abc755f0e7ffbcc2bbf79f22feae6/Tests/Integration/MSFT_xWebAppPool.config.ps1#L18)):*
Thanks heaps! :) @TysonJHayes - reckon it's OK for this one?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KFM7SjwUFvaF8mGXmWA:1009650786)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Ok, one final thing we've got to get changed: The parameter formatting.

Because we haven't been able to get a definitive decision on a style guideline on attribute format, we want to just make them consistent across all resources in a module.

Is it possible for you (and I know this is a complete pain) to change your parameter formats to match the other resource modules in xWebAdministration? e.g. [Type] $Name = 'abc' format?

I've let @TysonJHayes know that other than this I think it's all in really good shape and ready to merge.

Thanks again! Hope I'm not being too much of a PITA! :smile:

---

Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KFfa29ceqIe7l1NT95K:920590136)*
<!-- Sent from Reviewable.io -->

SNikalaichyk commented:
Hi, thanks a lot! Parameters have already been reformatted.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KFheMKSDEPxbiql_1pC:1599042043)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Awesome! Thank you heaps - it's all looking good to me!

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KFhuYlaRXIAE3oQnffo:-618639198)*
<!-- Sent from Reviewable.io -->

tysonjhayes commented:
Everything looks good to me, if you can just fix the merge conflicts I'll get this merged in. Thanks!

SNikalaichyk commented:
Merge conflicts should be resolved. Can you please restart AppVeyor build?

tysonjhayes commented:
I do not have such power, quickest way is to submit a small white space change (usually I just add a return to the yml file for appveyor) and submit.

@TravisEz13 or @KarolKaczmarek could either one of you restart this please?

KarolKaczmarek commented:
rerunning

SNikalaichyk commented:
Just a friendly reminder. Looking forward to seeing this merged.

PlagueHO commented:
@tysonjhayes  - looks like this is all good to go now :smile: CI build fixed!

mbreakey3 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 5 of 7 files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98#-:-KGIiWy1MszTV4Anz1qt:1417088721)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xwebadministration/98)*
<!-- Sent from Reviewable.io -->
